include/oneapi/dpl/pstl/algorithm_impl.h

```diff
@@ -2948,6 +2949,49 @@ __pattern_remove_if(__parallel_tag<_IsVector> __tag, _ExecutionPolicy&& __exec,
 // merge
 //------------------------------------------------------------------------
 
+template<typename It1, typename It2, typename ItOut, typename _Comp>
+std::pair<It1, It2>
+__brick_merge_2(It1 __it_1, It1 __it_1_e, It2 __it_2, It2 __it_2_e, ItOut __it_out, ItOut __it_out_e, _Comp __comp,
```

Comment:
Probably the existing implementation of `__serial_merge` is more faster then this.

Reply:
This version is for the all types of iterators (including  "forward" iterator).
The mentioned by you `__serial_merge` works just for random access ranges/iterators.

Reply:
Agree.
But I believe it can be rewrited for the forward iterators and works without indexes...

Reply:
There is also and a semantic discrepancy  - `__serial_merge` does not more then `__chunk` steps of serial merge.